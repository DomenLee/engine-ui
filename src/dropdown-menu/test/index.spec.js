import { mount, later } from '../../../test';

function renderWrapper(options = {}) {
  return mount({
    template: `
      <ghb-dropdown-menu :direction="direction" :close-on-click-outside="closeOnClickOutside">
        <ghb-dropdown-item v-model="value" :title="title" :options="options" />
        <ghb-dropdown-item v-model="value" :title="title" :options="options" />
      </ghb-dropdown-menu>
    `,
    data() {
      return {
        value: options.value || 0,
        title: options.title || '',
        direction: options.direction || 'down',
        closeOnClickOutside: options.closeOnClickOutside,
        options: [
          { text: 'A', value: 0, icon: options.icon },
          { text: 'B', value: 1, icon: options.icon },
        ],
      };
    },
  });
}

test('show dropdown item', async () => {
  const wrapper = renderWrapper();

  await later();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');

  titles.at(0).trigger('click');
  expect(wrapper).toMatchSnapshot();

  titles.at(1).trigger('click');
  expect(wrapper).toMatchSnapshot();

  titles.at(1).trigger('click');
  expect(wrapper).toMatchSnapshot();
});

test('render option icon', async () => {
  const wrapper = renderWrapper({
    icon: 'success',
  });

  await later();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');

  titles.at(0).trigger('click');
  expect(wrapper).toMatchSnapshot();
});

test('close-on-click-outside', async () => {
  const wrapper = renderWrapper({
    closeOnClickOutside: true,
  });

  await later();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');

  titles.at(0).trigger('click');
  document.body.click();
  expect(wrapper).toMatchSnapshot();
});

test('disable close-on-click-outside', async () => {
  const wrapper = renderWrapper({
    closeOnClickOutside: false,
  });

  await later();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');

  titles.at(0).trigger('click');
  document.body.click();
  expect(wrapper).toMatchSnapshot();
});

test('direction up', async () => {
  const { innerHeight } = window;
  window.innerHeight = 1000;

  const wrapper = renderWrapper({
    direction: 'up',
  });

  await later();
  expect(wrapper).toMatchSnapshot();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');
  titles.at(0).trigger('click');
  expect(wrapper).toMatchSnapshot();

  window.innerHeight = innerHeight;
});

test('click option', async () => {
  const wrapper = renderWrapper();

  await later();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');
  titles.at(0).trigger('click');

  const options = wrapper.findAll('.ghb-dropdown-item .ghb-cell');
  options.at(1).trigger('click');

  await later();
  expect(wrapper).toMatchSnapshot();
});

test('title prop', async () => {
  const wrapper = renderWrapper({ title: 'Title' });
  await later();
  expect(wrapper).toMatchSnapshot();
});

test('didn`t find matched option', async () => {
  const wrapper = renderWrapper({ value: -1 });
  await later();
  expect(wrapper).toMatchSnapshot();
});

test('destroy one item', async () => {
  const wrapper = mount({
    template: `
      <ghb-dropdown-menu>
        <ghb-dropdown-item v-if="render" v-model="value" :options="options" />
        <ghb-dropdown-item v-model="value" :options="options" />
      </ghb-dropdown-menu>
    `,
    data() {
      return {
        value: 0,
        render: true,
        options: [
          { text: 'A', value: 0 },
          { text: 'B', value: 1 },
        ],
      };
    },
  });

  await later();
  wrapper.setData({ render: false });
  expect(wrapper).toMatchSnapshot();
});

test('disable dropdown item', async () => {
  const wrapper = mount({
    template: `
      <ghb-dropdown-menu>
        <ghb-dropdown-item disabled v-model="value" :options="options" />
      </ghb-dropdown-menu>
    `,
    data() {
      return {
        value: 0,
        options: [
          { text: 'A', value: 0 },
          { text: 'B', value: 1 },
        ],
      };
    },
  });

  const title = wrapper.find('.ghb-dropdown-menu__title');
  title.trigger('click');
  expect(wrapper).toMatchSnapshot();
});

test('change event', async () => {
  const onChange = jest.fn();

  const wrapper = mount({
    template: `
      <ghb-dropdown-menu>
        <ghb-dropdown-item v-model="value" :options="options" @change="onChange" />
        <ghb-dropdown-item v-model="value" :options="options" />
      </ghb-dropdown-menu>
    `,
    data() {
      return {
        value: 0,
        options: [
          { text: 'A', value: 0 },
          { text: 'B', value: 1 },
        ],
      };
    },
    methods: {
      onChange,
    },
  });

  await later();

  const titles = wrapper.findAll('.ghb-dropdown-menu__title');
  titles.at(0).trigger('click');

  const options = wrapper.findAll('.ghb-dropdown-item .ghb-cell');
  options.at(0).trigger('click');

  expect(onChange).toHaveBeenCalledTimes(0);

  options.at(1).trigger('click');
  expect(onChange).toHaveBeenCalledWith(1);
  expect(onChange).toHaveBeenCalledTimes(1);
});

test('toggle method', async (done) => {
  const wrapper = mount({
    template: `
      <ghb-dropdown-menu>
        <ghb-dropdown-item ref="item" />
      </ghb-dropdown-menu>
    `,
    async mounted() {
      // show
      this.$refs.item.toggle(true, { immediate: true });
      await later();

      const content = wrapper.find('.ghb-dropdown-item__content');
      expect(content.element.style.display).toEqual('');

      // hide
      this.$refs.item.toggle(false, { immediate: true });
      await later();
      expect(content.element.style.display).toEqual('none');

      done();
    },
  });
});

test('title slot', () => {
  const wrapper = mount({
    template: `
      <ghb-dropdown-menu>
        <ghb-dropdown-item>
          <template #title>
            Custom Title
          </template>
        </ghb-dropdown-item>
      </ghb-dropdown-menu>
    `,
  });

  expect(wrapper).toMatchSnapshot();
});
