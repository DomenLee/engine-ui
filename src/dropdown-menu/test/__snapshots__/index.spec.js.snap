// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click option 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">B</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">B</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-overlay" style="z-index: 2013; animation-duration: 0.2s; position: absolute;" name="ghb-fade"></div>
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s; display: none; z-index: 2014;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>A</span></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>B</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`close-on-click-outside 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s; display: none;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`destroy one item 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <!---->
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`didn\`t find matched option 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis"></div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis"></div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`direction up 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--up" style="bottom: 0px; display: none;">
      <!---->
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--up" style="bottom: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`direction up 2`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--active"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--up" style="bottom: 1000px;">
      <div class="ghb-popup ghb-popup--bottom ghb-dropdown-item__content" style="transition-duration: 0.2s;" name="ghb-popup-slide-bottom">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--up" style="bottom: 1000px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`disable close-on-click-outside 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--active ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`disable dropdown item 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar">
    <div role="button" tabindex="-1" class="ghb-dropdown-menu__item ghb-dropdown-menu__item--disabled"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`render option icon 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--active ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active"><i class="ghb-icon ghb-icon-success ghb-cell__left-icon">
            <!----></i>
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option"><i class="ghb-icon ghb-icon-success ghb-cell__left-icon">
            <!----></i>
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`show dropdown item 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--active ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`show dropdown item 2`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title ghb-dropdown-menu__title--active ghb-dropdown-menu__title--down"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0s; display: none;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`show dropdown item 3`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar ghb-dropdown-menu__bar--opened">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">A</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0s; display: none;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px;">
      <div class="ghb-popup ghb-popup--top ghb-dropdown-item__content" style="transition-duration: 0.2s; display: none;" name="ghb-popup-slide-top">
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option ghb-dropdown-item__option--active">
          <div class="ghb-cell__title"><span>A</span></div>
          <div class="ghb-cell__value"><i class="ghb-icon ghb-icon-success ghb-dropdown-item__icon">
              <!----></i></div>
        </div>
        <div role="button" tabindex="0" class="ghb-cell ghb-cell--clickable ghb-dropdown-item__option">
          <div class="ghb-cell__title"><span>B</span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`title prop 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">Title</div></span></div>
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">Title</div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`title slot 1`] = `
<div class="ghb-dropdown-menu">
  <div class="ghb-dropdown-menu__bar">
    <div role="button" tabindex="0" class="ghb-dropdown-menu__item"><span class="ghb-dropdown-menu__title"><div class="ghb-ellipsis">
            Custom Title
          </div></span></div>
  </div>
  <div>
    <div class="ghb-dropdown-item ghb-dropdown-item--down" style="top: 0px; display: none;">
      <!---->
    </div>
  </div>
</div>
`;
